comp/otelcol/collector/impl-pipeline/pipeline.go

```diff
@@ -112,6 +123,166 @@ func (c *collectorImpl) stop(context.Context) error {
 	return nil
 }
 
+func (c *collectorImpl) fillFlare(fb flaretypes.FlareBuilder) error {
+	if !c.config.GetBool("otel.enabled") {
+		return nil
+	}
+
+	// request config from Otel-Agent
+	responseBytes, err := c.requestOtelConfigInfo(c.config.GetInt("otel.extension_url"))
+	if err != nil {
+		fb.AddFile("otel/otel-agent.log", []byte(fmt.Sprintf("did not get otel-agent configuration: %v", err))) //nolint:errcheck
+		return nil
+	}
+
+	// add raw response to flare, and unmarshal it
+	fb.AddFile("otel/otel-response.json", responseBytes) //nolint:errcheck
+	var responseInfo configResponseInfo
+	if err := json.Unmarshal(responseBytes, &responseInfo); err != nil {
+		fb.AddFile("otel/otel-agent.log", []byte(fmt.Sprintf("could not read sources from otel-agent response: %s", responseBytes))) //nolint:errcheck
+		return nil
+	}
+
+	fb.AddFile("otel/otel-flare/startup.cfg", []byte(toJSON(responseInfo.StartupConf)))     //nolint:errcheck
+	fb.AddFile("otel/otel-flare/runtime.cfg", []byte(toJSON(responseInfo.RuntimeConf)))     //nolint:errcheck
+	fb.AddFile("otel/otel-flare/environment.cfg", []byte(toJSON(responseInfo.Environment))) //nolint:errcheck
+	fb.AddFile("otel/otel-flare/cmdline.txt", []byte(responseInfo.Cmdline))                 //nolint:errcheck
+
+	// retrieve each source of configuration
+	for _, src := range responseInfo.Sources {
+		var err error
+		var response *http.Response
+		retries := 3
```

Comment:
Why are we expecting error from the OTEL endpoint ?

Reply:
To be honest it was an attempt to handle the flakiness in this test in the CI, but that doesn't seem to be fixing it. I have removed it.